src/filters/token_router.rs

```diff
@@ -111,18 +112,18 @@ impl Filter for TokenRouter {
                 self.metrics.packets_dropped_no_token_found.inc();
                 None
             }
-            Some(value) => match value.downcast_ref::<Vec<u8>>() {
-                Some(token) => match ctx
+            Some(value) => match value {
+                Value::Bytes(token) => match ctx
                     .endpoints
-                    .retain(|e| e.metadata.known.tokens.contains(token))
+                    .retain(|e| e.metadata.known.tokens.contains(&**token))
                 {
                     RetainedItems::None => {
                         self.metrics.packets_dropped_no_endpoint_match.inc();
                         None
                     }
                     _ => Some(ctx.into()),
                 },
-                None => {
+                _ => {
                     if self.metrics.packets_dropped_invalid_token.get() % LOG_SAMPLING_RATE == 0 {
                         error!(
                             count = ?self.metrics.packets_dropped_invalid_token.get(),
```

Comment:
the error message here has a hardcoded `Vec<u8>` which is now out  of date

Reply:
I'll update the error message, but I wonder if we should be logging an error here, instead of panicking and terminating the process. Since the type not matching would be the result of programmer error, not application error.

Like would there ever be a situation where the type would be correct 90% of the time? It seems like it would either, always be correct or always be wrong.

Reply:
> panicking and terminating the process

Maybe I'm missing something, but I don't see a panic here? We drop the packet if the type is not correct, and do a (sampled) log.

Reply:
@markmandel Right, I'm saying maybe we **should panic** here, because it would a programmer error.

Reply:
yeah its most likely a programmer error (it could also be a buggy control plane e.g after/during an upgrade in which case we might want to keep running so that a new filterchain update is eventually delivered to the proxy) but I don't see a need to panic as a result? we'll end up dropping the packet and logging and metrics. Compared to those, panicking seems harsh and we'd need to document the behavior too which doesn't seem worth it